room-battle.js

```diff
@@ -847,6 +848,8 @@ class Battle {
 			// @ts-ignore
 			this.room.title = `${this.p1.name} vs. ${this.p2.name}`;
 			this.room.send(`|title|${this.room.title}`);
+			// @ts-ignore
+			if (!wasStarted && this.started) Rooms.global.onCreateBattleRoom(Users(this.p1.userid), Users(this.p2.userid), this.room, {rated: this.rated});
```

Comment:
Why not just run this directly after `this.started = true` in `makePlayer`? Then you wouldn't need the temporary variable.

Reply:
Also don't forget: Remove the now-redundant `onCreateBattleRoom` call in `rooms.js` or wherever it is.